src/phoenix/experimental/evals/models/openai.py

```diff
@@ -0,0 +1,128 @@
+    """Model name to use."""
+    temperature: float = 0.7
+    """What sampling temperature to use."""
+    max_tokens: int = 256
+    """The maximum number of tokens to generate in the completion.
+    -1 returns as many tokens as possible given the prompt and
+    the models maximal context size."""
+    top_p: float = 1
+    """Total probability mass of tokens to consider at each step."""
+    frequency_penalty: float = 0
+    """Penalizes repeated tokens according to frequency."""
+    presence_penalty: float = 0
+    """Penalizes repeated tokens."""
+    n: int = 1
+    """How many completions to generate for each prompt."""
+    model_kwargs: Dict[str, Any] = field(default_factory=dict)
+    """Holds any model parameters valid for `create` call not explicitly specified."""
+    batch_size: int = 20
+    # TODO: IMPLEMENT BATCHING
+    """Batch size to use when passing multiple documents to generate."""
+    request_timeout: Optional[Union[float, Tuple[float, float]]] = None
+    """Timeout for requests to OpenAI completion API. Default is 600 seconds."""
+    max_retries: int = 6
+    """Maximum number of retries to make when generating."""
+    retry_min_seconds: int = 10
+    """Minimum number of seconds to wait when retrying."""
+    retry_max_seconds: int = 60
+    """Maximum number of seconds to wait when retrying."""
+
+    def __post_init__(self) -> None:
+        if self.openai_api_key is None:
+            api_key = os.getenv(OPENAI_API_KEY_ENVVAR_NAME)
+            if api_key is None:
+                # TODO: Create custom AuthenticationError
+                raise RuntimeError(
+                    "OpenAI's API key not provided. Pass it as an argument to 'openai_api_key' "
+                    "or set it in your environment: 'export OPENAI_API_KEY=sk-****'"
+                )
+            self.openai_api_key = api_key
+
+    def _generate(self, prompt: str, instruction: Optional[str]) -> str:
+        invoke_params = self.invocation_params
+        messages = [{"role": "user", "content": prompt}]
+        if instruction:
+            messages.append({"role": "system", "content": instruction})
+        response = self._generate_with_retry(
+            messages=messages,
+            **invoke_params,
+        )
+        # TODO: This is a bit rudimentary, should improve
+        resp_text = str(response["choices"][0]["message"]["content"]).strip()
+        return resp_text
+
+    def _generate_with_retry(self, **kwargs: Any) -> Any:
+        """Use tenacity to retry the completion call."""
+        retry_decorator = create_base_retry_decorator(
+            error_types=OPENAI_RETRY_ERRORS,
+            min_seconds=self.retry_min_seconds,
+            max_seconds=self.retry_max_seconds,
+            max_retries=self.max_retries,
+        )
+
+        @retry_decorator
+        def _completion_with_retry(**kwargs: Any) -> Any:
+            return openai.ChatCompletion.create(**kwargs)  # type:ignore
```

Comment:
I'm not 100% sure, but this looks like it is using the synchronous chat completion API. The OpenAI Python SDK has a separate function for async function calls. https://github.com/openai/openai-python#async-api. Can you verify that this is doing async calls?